### PR TITLE
refactor: use tokens from dab

### PR DESCRIPTION
**Pull request by aleacevedo**
[WIP] refactor: use tokens from dab

### CODE REVIEW

File: src/utils/storage/update_handlers/v0.14.2.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+                    symbol: asset.symbol,
+                    canisterId: asset.canisterId,
+                    standard:  wallet.registeredTokens[asset.canisterId]?.standard || TOKENS[asset.symbol].standard,
+                    decimals: wallet.registeredTokens[asset.canisterId]?.decimals || TOKENS[asset.symbol].decimals,
+                    color: wallet.registeredTokens[asset.canisterId]?.color
+                }
+            }}), {})
+        }))
+    }
+});
```
Comment (tomiir) [inline src/utils/storage/update_handlers/v0.14.2.ts:23]: time to test this out lol

File: src/PlugWallet/index.ts [hunk 2]
```diff
@@ -388,12 +343,36 @@ class PlugWallet {
...
+    if (canisterId === TOKENS.XTC.canisterId) {
+      try {
+        if ('transactionId' in result) {
+          const trxId = result.transactionId;
+          await requestCacheUpdate(this.principal, [BigInt(trxId)]);
+        }
+      } catch (e) {
+        console.log('Kyasshu error', e);
+      }
+    }
```
Comment (tomiir) [inline src/PlugWallet/index.ts:373]: shuoldnt this be in XTC's send method?

Comment (aleacevedo) [reply src/PlugWallet/index.ts:373]: requestCacheUpdate is 100% plug stuff

File: src/utils/storage/update_handlers/v0.14.2.ts (hunk 1 above)
Comment (aleacevedo) [reply src/utils/storage/update_handlers/v0.14.2.ts:23]: image

File: src/PlugKeyRing/index.ts [hunk 3]
```diff
@@ -136,7 +133,10 @@ class PlugKeyRing {
     } = ((await this.storage.get()) || {}) as StorageData;
     if (isInitialized && vault) {
       const newVersion = getVersion();
-      const decrypted = newVersion === version ? this.decryptState(vault, password) : handleStorageUpdate(version, this.decryptState(vault, password));
+      const decrypted =
+        newVersion === version
+          ? this.decryptState(vault, password)
+          : handleStorageUpdate(version, this.decryptState(vault, password));
```
Comment (tomiir) [inline src/PlugKeyRing/index.ts:139]: you should call decryptState only once I guess.

File: src/utils/storage/update_handlers/v0.14.5.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            canisterId: asset.canisterId,
+            standard:
+              wallet.registeredTokens[
+                asset.canisterId
+              ]?.standard.toUpperCase() || TOKENS[asset.symbol].standard,
+            decimals:
+              wallet.registeredTokens[asset.canisterId]?.decimals ||
+              TOKENS[asset.symbol].decimals,
+            color: wallet.registeredTokens[asset.canisterId]?.color,
+          },
```
Comment (tomiir) [inline src/utils/storage/update_handlers/v0.14.5.ts:25]: have you done upgrade tests on this?

**Review by tomiir — APPROVED** [review]

**Pull request merged**